File: RecyclerView/app/src/main/java/com/coder178/recyclerview/MainActivity.java
package com.coder178.recyclerview;

import androidx.appcompat.app.AppCompatActivity;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import android.os.Bundle;

public class MainActivity extends AppCompatActivity {

    RecyclerView recyclerView;
    String[] str = {"item1","item2","item3","item3","item4","item5","item6","item7","item8","item9","Item10"};
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        recyclerView = findViewById(R.id.recyclerView);
        recyclerView.setLayoutManager(new LinearLayoutManager(this));
        CustomAdapter c = new CustomAdapter(str);
        recyclerView.setAdapter(c);
    }
}

File: CustomListView/app/src/main/java/com/example/myapplication/MainActivity.kt
package com.example.myapplication

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.ListView
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {

    private lateinit var expensesList: ListView
    private val expenses = ArrayList<Expense>()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        with(expenses) {
            add(Expense("Groceries", 5000f))
            add(Expense("Transportation", 8000f))
            add(Expense("Rent", 50000f))
            add(Expense("Cell Phone", 800f))
            add(Expense("Utility Bills", 6500f))
            add(Expense("Insurance", 5000f))
        }

        expensesList = findViewById(R.id.expenses_lists)
        val expenseAdapter = CustomExpenseAdapter(this, expenses)
        expensesList.adapter = expenseAdapter

        total.text = Expense.total(expenses)
    }
}
